### PR TITLE
fix(Button): fix negligent selectors

### PR DESCRIPTION
**Pull request by codecaaron**
### Overview

Fixes up the folly of my negligent wildcard use.   Current usage doesn't have markup as children but it will and this breaks that.  I moved the margin changes to a wrapper specifically for IconButton.

- Isolated the margin offset to maintain dimensions to icon button.  Rest stays vertical align middle.

### PR Checklist

- [ ] Related to designs:
- [] Related to JIRA ticket: ABC-123
- [x] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@10.1.7-alpha.0bb41c.0<br/>@codecademy/gamut@25.2.3-alpha.0bb41c.0<br/>@codecademy/styleguide@29.3.6-alpha.0bb41c.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://603545dbc303240ad04967a9--gamut-preview.netlify.app>https://603545dbc303240ad04967a9--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/603545dbc303240ad04967a9>Deploy Logs</a></p>

File: packages/gamut/src/Button/ButtonInner.tsx [hunk 1]
```diff
@@ -30,13 +30,6 @@ export const ButtonInner = styled('span', {
 
     > * {
       vertical-align: middle;
-
-      &:first-of-type {
-        margin-left: -1px;
-      }
-      &:last-of-type {
-        margin-right: -1px;
-      }
```
Comment (codecaaron) [inline packages/gamut/src/Button/ButtonInner.tsx:39]: I don't know what I was thinking putting this here. 

Ostensibly this is fine with `:first-child` + `:last-child` to correct the offset for the spacers but this really had no business being in a general spot.

**Review by jakemhiller — APPROVED** [review]

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@10.1.7-alpha.0455c1.0<br/>@codecademy/gamut@25.2.3-alpha.0455c1.0<br/>@codecademy/styleguide@29.3.6-alpha.0455c1.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60354d68f9f10a140031f311--gamut-preview.netlify.app>https://60354d68f9f10a140031f311--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60354d68f9f10a140031f311>Deploy Logs</a></p>

**Review by JoshuaKGoldberg — APPROVED** [review]
Haha I thought the old way was acceptable because it was limited to `ButtonInner`? But sure, this looks nice too!

**Comment by codecaaron** [comment]
> Haha I thought the old way was acceptable because it was limited to `ButtonInner`? But sure, this looks nice too!

Yeah I thought it'd be fine if we were just using text.  But that was a naive assumption, though I would like to know why we need arbitrary mark up down stream in these 🤷 .

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@10.1.8-alpha.f90a76.0<br/>@codecademy/gamut@25.3.1-alpha.f90a76.0<br/>@codecademy/styleguide@29.4.1-alpha.f90a76.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://603550def9f10a140031f614--gamut-preview.netlify.app>https://603550def9f10a140031f614--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/603550def9f10a140031f614>Deploy Logs</a></p>

**Pull request merged**